Ironmon-Tracker.lua

```diff
@@ -418,6 +420,28 @@ function Main.GenerateNextRom()
 	}
 end
 
+function Main.SaveCurrentRom(filename)
+	local newseedfilename = filename:gsub(Constants.Files.PostFixes.AUTORANDOMIZED, Constants.Files.PostFixes.PREVIOUSATTEMPT)
+	Main.CopyFile(filename, newseedfilename)
+	Main.CopyFile(string.format("%s.log", filename), string.format("%s.log", newseedfilename))
+end
+
+function Main.CopyFile(filename, newfilename)
+	if Main.FileExists(filename) then
+		local original = io.open(filename, "rb")
+		local copy = io.open(newfilename, "wb")
+
+		local nextBlock = original:read(2^13)
+		while nextBlock ~= nil do
+			copy:write(nextBlock)
+			nextBlock = original:read(2^13)
+		end
+
+		original:close()
+		copy:close()
+	end
+end
```

Comment:
I want to second Fellshadow's concern here that the two files that are being opened should check for `nil` similar to how all other `io.open` calls are handled. While it's a minor change overall, simply having any nil conflict often causes the tracker script to crash, which in turn many end-users don't often understand what happens or know how to fix it.

I'd recommend dropping the `Main.FileExists(filename)` and add the checks after opening the files:

> if original == nil or copy == nil then
> print(string.format("Error: Unable to create a backup copy of %s."), filename)
> return
> end